mlir/lib/AsmParser/LocationParser.cpp

```diff
@@ -187,5 +210,8 @@ ParseResult Parser::parseLocationInstance(LocationAttr &loc) {
     return success();
   }
 
+  if (getToken().getSpelling() == "dialect")
```

Comment:
Let me try that. I think I did, and it didn't work, but I'm not sure why anymore.

Reply:
I found why :P OperationParser::parseLocationAlias manually checks the spelling looking for a dialect because it's looking for hash aliases. I think I can re-work it though

Reply:
Updated - got it working :) It invalidated a test that was checking for an error in a case that actually would now be valid if the dialect was registered.